File: DecompiledGameScripts/core/components/scriptComponents/kurtzComponent.swift

public class KurtzComponent extends ScriptableComponent {

  private final func OnGameAttach() -> Void;

  protected cb func OnAIEvent(aiEvent: ref<AIEvent>) -> Bool {
    switch aiEvent.name {
      case n"ScannerOn":
        GameObject.StartReplicatedEffectEvent(this.GetOwner(), n"scanner");
        break;
      case n"ScannerOff":
        GameObject.BreakReplicatedEffectLoopEvent(this.GetOwner(), n"scanner");
    };
  }

  private final func OnGameDetach() -> Void {
    GameObject.BreakReplicatedEffectLoopEvent(this.GetOwner(), n"scanner");
  }

  protected cb func OnDefeated(evt: ref<DefeatedEvent>) -> Bool {
    GameObject.BreakReplicatedEffectLoopEvent(this.GetOwner(), n"scanner");
  }

  protected cb func OnDeath(evt: ref<gameDeathEvent>) -> Bool {
    GameObject.BreakReplicatedEffectLoopEvent(this.GetOwner(), n"scanner");
  }
}

File: Definitions/Game_Definitions/classes/KurtzComponent.lua
---@meta
---@diagnostic disable

---@class KurtzComponent : gameScriptableComponent
KurtzComponent = {}

---@return KurtzComponent
function KurtzComponent.new() return end

---@param props table
---@return KurtzComponent
function KurtzComponent.new(props) return end

---@param aiEvent AIAIEvent
---@return Bool
function KurtzComponent:OnAIEvent(aiEvent) return end

---@param evt gameeventsDeathEvent
---@return Bool
function KurtzComponent:OnDeath(evt) return end

---@param evt gameeventsDefeatedEvent
---@return Bool
function KurtzComponent:OnDefeated(evt) return end

function KurtzComponent:OnGameAttach() return end

function KurtzComponent:OnGameDetach() return end

